src/QMCHamiltonians/ObservableHelper.h

```diff
@@ -35,10 +35,11 @@ using value_type = QMCTraits::FullPrecRealType;
 class ObservableHelper
 {
 public:
+  ObservableHelper() = delete;
   /**
-   * default constructor
+   * Favored constructor
+   * \param[in] title         is the ordered hdf5 group path elements of the observable
    */
-  ObservableHelper(const std::string& title = "dummy");
```

Comment:
I actually prefer getting ride of the constructor with `std::vector<std::string>`, I rarely saw the need of vector and saw more cumbersome caused by vector.

can we just have
```
ObservableHelper(const std::string& title);
```
Then we can use `emplace_back("abc")` instead of `push_back({{"abc"}})`

Reply:
The point of #4322 was to remove that state games that made the single string constructor useful.  Basically the library was keeping that group path due to the sequence of calls from the different observables.  Most of the ported estimators use a multi element path.  Several hamiltonian operators just dump every hdf5 dataset into the global group with some gloppy name and should be using the `{name_, "variable_name"}`